pkg/exec/lint.go

```diff
@@ -63,17 +63,20 @@ func (m *Mixin) Lint() (linter.Results, error) {
 				continue
 			}
 
+			// Derive key to be used to locate coordinates in the manifest.
+			// The embedded bash flag, 'c' is not unique enough,
+			// so set to the unique step description, or, if values non-empty,
+			// the first value
+			key := step.Description
```

Comment:
I'm concerned that this solution doesn't guarantee that the line number is correct. Given a non-unique description (porter doesn't require unique descriptions, and won't in the future), and the same lint error multiple times in the file, this could identify the same lines over and over again instead of different lines throughout the file.

Let's step back and see if we can find a way to identify the line number with confidence. This could involve a couple solutions, maybe you see more?

1. the mixin reports a relative line number based on the raw text input given. Then porter uses the context of the full manifest to turn the relative line number into an absolute line number.
2. Porter starts passing in different data to build, either the full manifest, or structured data with line numbers so that the mixin can report the absolute line number itself.

Reply:
I added some notes on my current thinking/solution for the scenario where there are duplicate offending keys that trigger the linter.  WDYT?  My first impulse was to go this approach as it seemed the lighter-weight option (as opposed to refactoring to pass in add'l (potentially structured) data into the mixin query)... but we can definitely go another route if this latest iteration still has issues.